### ISSUE TITLE
[Request] Add (Optional) Trusted Header for Remote IP

### ISSUE THREAD

**Issue by dwmunster** [go-joe/http-server#3]
## Summary

Currently, the [debug logging functionality](https://github.com/go-joe/http-server/blob/20a7552207d436c2c9dcd20679c8df206d35d5ea/server.go#L100) reports the IP address assigned to the `RemoteAddr` field. In all but the most simple case, this field will likely point to part of the user's infrastructure, such as a reverse proxy or load balancer. Adding the ability to specify a trusted header will allow for more complete logging.

## Basic example

```go
joe.New(..., http.Server(":80", WithTrustedHeader("x-forwarded-for")))
````

## Motivation

After observing an automated scan for vulnerabilities in my logs, I noticed that the reported remote IP all point to my reverse proxy. Since that is not particularly actionable, it would be nice to report an actual client IP.

## Potential Issues

* We cannot simply search through a list of commonly defined headers because a malicious user could set their own values. Therefore, it is necessary to specify which header (if any) should be trusted to contain the real IP address of the client.
* Since there could be a series of reverse proxies, load balancers, etc. in front of the server, the trusted field could contain a [comma and space](https://en.wikipedia.org/wiki/X-Forwarded-For#Format) list of IP addresses. We could report back the entire header value or (perhaps more usefully) simply the first.
* Note: the capitalization of the header can be arbitrary, the header names are normalized by `request.Header.Get`.

## 
I plan on implementing a PR for this very soon and will add the link once I have done so.

**Comment by dwmunster** [comment]
I'm copying some of the comments from #4 here so that any decisions are captured in one place.

> > Since there could be a series of reverse proxies, load balancers, etc. in front of the server, the trusted field could contain a comma and space list of IP addresses. We could report back the entire header value or (perhaps more usefully) simply the first.
> 
> Currently if the header is a list of addresses, will `net.JoinHostPort` still work? Should we have a unit test for this case as well?


Looks like it should behave reasonably: https://play.golang.org/p/VVNn-oLHU2g
`"127.0.0.1, 192.168.0.1"` becomes `"127.0.0.1, 192.168.0.1:80"`. While the first entry is the most useful, the port only refers to the final connection. That leaves two options:

1. Return `"127.0.0.1, 192.168.0.1:80"`. This has the most information, though much of it shouldn't be a surprise. Furthermore, it may confuse someone/tools expecting a single IP address (though it will log just fine).
2. Return `"127.0.0.1"`. We will have no information about what port they are connected on, so we can't forward that. This contains the most relevant piece of information, but may confuse tools expecting it to look like the output from `req.RemoteAddr` since it will not have a port number.

In either case, we should add a test to capture that expectation. I personally lean towards 2, but don't have a strong opinion.

**Comment by fgrosse** [comment]
I also do not have a strong opinion on the matter. To decide if its ok to have a comma separated list in the `RequestEvent.RemoteAddr` field I looked up the definition of the corresponding `http.Request` field which we currently use to fill that field:

```go
// RemoteAddr allows HTTP servers and other software to record
// the network address that sent the request, usually for
// logging. This field is not filled in by ReadRequest and
// has no defined format. The HTTP server in this package
// sets RemoteAddr to an "IP:port" address before invoking a
// handler.
// This field is ignored by the HTTP client.
RemoteAddr string
```

So with that in mind I am not too concerned about confusing tools because we already do not make big promises about the format and the primary use case is logging. I would roll with whatever fits your current use case the best :sunglasses:

**Comment by fgrosse** [comment]
Feature is released via `v0.5.0` :tada: :tada: :tada:
